File: challenges/scripts/Topics/number_subscriber.py
#!/usr/bin/env python3

import rospy
from std_msgs.msg import Int64

def call_back_number(msg):
    rospy.loginfo(msg)

if __name__ == "__main__":
    rospy.init_node('number_subscriber')

    sub = rospy.Subscriber("/number_counter", Int64, call_back_number)

    rospy.spin()

File: basics/Topics/src/number_subscriber.cpp
#include <ros/ros.h>
#include <std_msgs/Int64.h>

void	callback_number(const std_msgs::Int64 &msg)
{
	ROS_INFO("Data: %d", (int)msg.data);
}

int	main(int argc, char **argv)
{
	ros::init(argc, argv, "Number_Subscriber");
	ros::NodeHandle nh;
	ros::Subscriber sub = nh.subscribe("/number_counter", 1000, callback_number);

	ros::spin();
}
